src/Symfony/Component/Form/Extension/Core/Type/MultiStepType.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class MultiStepType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setRequired('steps')
+            ->setDefault('current_step', static function (Options $options): string {
+                /** @var array<string, mixed> $steps */
+                $steps = $options['steps'];
+                $firstStep = array_key_first($steps);
+                if (!\is_string($firstStep)) {
+                    throw new \InvalidArgumentException('The option "steps" must be an associative array.');
+                }
+
+                return $firstStep;
+            });
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $currentStep = $options['steps'][$options['current_step']];
```

Comment:
As part of the `current_step` value validation (`setAllowedValues`) we should throw a friendly exception if it's not included in the keys of the steps array

Reply:
Sorry, in this case you should use `setNormalizer` instead of `setAllowedValues` as the validation would depend on another option value